### DISCUSSION TITLE
Reset Api also resets the handle form data creating duplicates

### DISCUSSION THREAD

**Discussion by HashiW** [react-hook-form/react-hook-form#2770] [Q&A]
Using the reset api in the use Effect adds to the data object in handle submit. On submit it creates duplicates in my database. How to reset the form without adding it to the data object in handle Submit?

**Comment by bluebill1049** [comment]
codesandbox?

**Comment by HashiW** [comment]
[https://codesandbox.io/s/practical-black-tbo5n?file=/src/App.js](). Reset Api also resets the formData as seen through the console log. 

In my app I make a GET request then add this data to the reset. However on submit I end up sending the same data from the GET request back to the database causing duplicates to occur. How would I stop this from occuring?

**Reply by bluebill1049** [reply]
you should filter the data in your onSubmit function, RHF produce what's in the form.

**Reply by HashiW** [reply]
Is there a way to not add the reset directly to the Data object? A use case would be using reset for persistence.  I'm only trying to persist form data.

**Reply by bluebill1049** [reply]
question: why do you need to reset the data? 

it's going to be tricky because RHF only collects what's in the form. There is no smartness in it.

**Reply by HashiW** [reply]
In my dynamic form I cannot have duplicates. By resetting I query the users previous input and display it in the form. However that initial query gets resubmitted along with the newly inputted fields by the user creating duplicates. because reset updates the data object. 


 A solution I have now is to make certain fields unique in my postgres database which works however the user cannot update the dynamic field because parts of it cannot be resubmitted due to the unique constraint  

All I want to do  is in the Reset of a field array to remove the queried data from being resubmitted.

**Reply by HashiW** [reply]
Is there a way to know if a field has been altered. then I can submit only those fields that have been changed/touched.

**Reply by bluebill1049** [reply]
formState: touched? 
formState: dirtyFields?
